### PR TITLE
Upgrade to Java 21 and Spring Boot 3

### PR DESCRIPTION
**Pull request by WadeBarnes**
# Description

This PR includes the following proposed change(s):
- Upgrade project and development environment to Java 21
- Upgrade to Spring Boot 3.x
- Bump jag-cdds version
- Add dependabot configuration to help manage dependencies.
- Update all GitHub Actions - by dependabot
- Update project dependencies based on update recommendations from dependabot
- Address all Critical and High vulnerabilities identified by dependabot and Trivy

Update Trivy workflow
- Set version on the `aquasecurity/trivy-action` so dependabot can manage the updates.
- Set `limit-severities-for-sarif: true` so Trivy does not fail for Medium and Low vulnerabilities.

Update documentation
- Add section on how to perform vulnerability scanning locally.

## Type of change
- Java and related dependency upgrades.
- Workflow updates
- Documentation updates

## How Has This Been Tested?
- Running the project's unit tests.
- Ensuring the automated tests pass for the PR.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/test/java/ca/bc/gov/open/cdds/CourtControllerTests.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public void getDigitalDisplayCourtList() throws IOException {
         a.setCourtClassCd("A");
         a.setFileNumberTxt("A");
         a.setCounselFullNm("A");
-        clr.setAppearance(Collections.singletonList(a));
+        // clr.setAppearance(Collections.singletonList(a));
```
Comment (WadeBarnes) [inline src/test/java/ca/bc/gov/open/cdds/CourtControllerTests.java:79]: @jianmingtu, @KFloodBCGov, As discussed, this is the only change I had to make that was a little concerning and unexpected.  It appears the Appearance class has completely changed and does not contain a getter or setter for Appearance at all anymore.

@jianmingtu, Is looking into this.

**Comment by WadeBarnes** [comment]
It's also worth noting that since the upgrade the following schema validation errors occur, but don't seem to affect the build or artifacts.  @jianmingtu indicated he's seen this with the other ORDS projects too.

The validation stops at the first error.  The issue is not specific to the `dx:AgencyIdentifierType` type definition.  If that is commented out, the validation fails at the next complex type.

It would be nice to figure this out, because it wasn't happening before the upgrade and should not be happening now.

```
[INFO] --- jaxb30:0.15.0:generate (default) @ jag-cdds ---
[INFO] Sources are not up-to-date, XJC will be executed.
[ERROR] Error while parsing schema(s).Location [ file:/workspaces/jag-cdds/src/main/resources/xsdSchemas/cdds-models-1.xsd{9,108}].
org.xml.sax.SAXParseException: src-resolve: Cannot resolve the name 'dx:AgencyIdentifierType' to a(n) 'type definition' component.
    at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException (ErrorHandlerWrapper.java:204)
    at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.error (ErrorHandlerWrapper.java:135)
    at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError (XMLErrorReporter.java:396)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.reportSchemaErr (XSDHandler.java:4254)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.reportSchemaError (XSDHandler.java:4237)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.getGlobalDecl (XSDHandler.java:1728)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDElementTraverser.traverseNamedElement (XSDElementTraverser.java:405)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDElementTraverser.traverseLocal (XSDElementTraverser.java:194)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.traverseLocalElements (XSDHandler.java:3707)
    at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.parseSchema (XSDHandler.java:666)
    at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.loadSchema (XMLSchemaLoader.java:618)
    at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.loadGrammar (XMLSchemaLoader.java:577)
    at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.loadGrammar (XMLSchemaLoader.java:543)
    at com.sun.org.apache.xerces.internal.jaxp.validation.XMLSchemaFactory.newSchema (XMLSchemaFactory.java:281)
    at com.sun.tools.xjc.reader.xmlschema.parser.SchemaConstraintChecker.check (SchemaConstraintChecker.java:81)
    at org.jvnet.mjiip.v_3_0.XJC30Mojo.loadModel (XJC30Mojo.java:87)
    at org.jvnet.mjiip.v_3_0.XJC30Mojo.doExecute (XJC30Mojo.java:53)
    at org.jvnet.mjiip.v_3_0.XJC30Mojo.doExecute (XJC30Mojo.java:35)
    at org.jvnet.jaxb2.maven2.RawXJCMojo.doExecute (RawXJCMojo.java:568)
    at org.jvnet.jaxb2.maven2.RawXJCMojo.execute (RawXJCMojo.java:367)
    at org.jvnet.jaxb2.maven2.RawXJC3Mojo.execute (RawXJC3Mojo.java:27)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:904)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:281)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:204)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[ERROR] SchemaConstraintChecker failed
```

**Comment by WadeBarnes** [comment]
@jianmingtu, @KFloodBCGov, I just added a commit to upgrade to Java 21.

File: src/test/java/ca/bc/gov/open/cdds/CourtControllerTests.java (hunk 1 above)
Comment (WadeBarnes) [reply src/test/java/ca/bc/gov/open/cdds/CourtControllerTests.java:79]: @jianmingtu, I have a fix for this coming.

Comment (WadeBarnes) [reply src/test/java/ca/bc/gov/open/cdds/CourtControllerTests.java]: Issue resolved.

**Review by jianmingtu — APPROVED** [review]
looks good

**Comment by WadeBarnes** [comment]
Thanks @jianmingtu.  I need to coordinate the merge of https://github.com/bcgov/jag-cdds/pull/50 before this one gets merged.  That will require a little rebasing and conflict resolution that I've already tested locally.

**Comment by WadeBarnes** [comment]
The changes on the `release/1.1` branch have been merged.  This PR has been rebased on those changes and the merge conflicts have been resolved.

**Review by sturufous — APPROVED** [review]

**Pull request merged**